zebra-network/src/protocol/external/codec.rs

```diff
@@ -480,10 +483,11 @@ impl Codec {
             nonce: Nonce(reader.read_u64::<LittleEndian>()?),
             user_agent: String::zcash_deserialize(&mut reader)?,
             start_height: block::Height(reader.read_u32::<LittleEndian>()?),
-            relay: match reader.read_u8()? {
-                0 => false,
-                1 => true,
-                _ => return Err(Error::Parse("non-bool value supplied in relay field")),
+            relay: match reader.read_u8() {
+                Ok(val @ 0..=1) => val == 1,
+                Ok(_) => return Err(Error::Parse("non-bool value supplied in relay field")),
+                Err(err) if err.kind() == std::io::ErrorKind::UnexpectedEof => true,
+                Err(err) => Err(err)?,
```

Comment:
Please document that the `relay` field is optional, and link to the protocol reference:
https://developer.bitcoin.org/reference/p2p_networking.html#version

It might also be worth noting that `zcashd` only requires up to `address_recv`:
https://github.com/zcash/zcash/blob/11d563904933e889a11d9685c3b249f1536cfbe7/src/main.cpp#L6490-L6507

(But this is from pre-2010 Bitcoin, so we don't need to be compatible with it, and it is out of scope.)

Reply:
Added in [`10c405c`](https://github.com/ZcashFoundation/zebra/pull/5835/commits/10c405ca251398f016e76909ecc970294bcb43d6).